devito/ir/iet/nodes.py

```diff
@@ -888,8 +888,11 @@ def hoistable(self):
         return ()
 
     @property
-    def is_Useless(self):
-        return USELESS in self.properties
+    def useless(self):
```

Comment:
the compiler solves a complex problem -- generating halo exchange routines at the optimum depth in the code tree -- through a sequence of _simpler_ subtasks

one such subtaks creates HaloSpots, ie nodes where **potentially** a halo exchange is required. Ie, first the compiler acts conservatively, assuming a halo exchange is require as soon as a stencil expression is encountered (e.g., ``u[t, x-1]``)

later on, another subtask uses data dependence analysis to check that a halo exchange is **actually** required. Whey might it not be required? well, for example if the value of ``u`` hasn't changed since the last halo update. These useless HaloSpots are marked as "USELESS".

A final subtask, in the DLE, searches for all USELESS HaloSpots and drops them